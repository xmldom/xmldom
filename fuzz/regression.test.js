'use strict';
// wallaby:file.skip since wallaby is not concerned with fuzz testing
const { describe, expect, test, beforeAll } = require('@jest/globals');

const fs = require('fs');
const path = require('path');
const TARGETS = fs
	.readdirSync(__dirname)
	.filter((file) => file.endsWith('.target.js'))
	.map((target) => [target, path.join(__dirname, target)]);

TARGETS.forEach(([target, targetPath]) => {
	describe(target, () => {
		const regressionDir = path.join(__filename.replace(/\.js$/, ''), target);
		const testfiles = fs.readdirSync(regressionDir);
		const module = require(targetPath);
		beforeAll(() => {
			expect(fs.existsSync(regressionDir)).toBe(true);
			expect(testfiles.length).toBeGreaterThan(0);
		});
		testfiles.forEach((testfile) => {
			test(path.basename(testfile), () => {
				expect(module.fuzz(fs.readFileSync(path.join(regressionDir, testfile)))).toMatchSnapshot();
			});
		});
	});
});
