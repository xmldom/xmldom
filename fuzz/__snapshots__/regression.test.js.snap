// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dom-parser.html.target.js crash-doctype-in-element.xml 1`] = `-1`;

exports[`dom-parser.html.target.js crash-proto-prefix.xml 1`] = `-1`;

exports[`dom-parser.xml.target.js crash-doctype-in-element.xml 1`] = `-1`;

exports[`dom-parser.xml.target.js crash-proto-prefix.xml 1`] = `-1`;
